From 6c40fda47f5a081e79fee04308f003225c15c90c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 11 May 2020 15:16:56 -0700
Subject: [PATCH 1/2] Disable Werror

4.9 does not compile cleanly with gcc9+, therefore disable the
warnings as error, so we can ignore the additional new warnings turning
into errors and breaking build, ideally each one should be addressed in
code

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile                                | 2 +-
 drivers/net/wireless/rtl8812au/Makefile | 2 ++
 net/netfilter/ipvs/Makefile             | 2 ++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c2d679059c2d..8215d1089339 100644
--- a/Makefile
+++ b/Makefile
@@ -393,7 +393,7 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common -fshort-wchar \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
-		   -Werror	\
+		   -Wno-error	\
 		   -std=gnu89 $(call cc-option,-fno-PIE)
 
 
diff --git a/drivers/net/wireless/rtl8812au/Makefile b/drivers/net/wireless/rtl8812au/Makefile
index 2844a37e716e..696d4d0fcbfd 100644
--- a/drivers/net/wireless/rtl8812au/Makefile
+++ b/drivers/net/wireless/rtl8812au/Makefile
@@ -13,6 +13,8 @@ EXTRA_CFLAGS += -Wno-unused-variable
 #EXTRA_CFLAGS += -Wno-unused-label
 EXTRA_CFLAGS += -Wno-unused-parameter
 EXTRA_CFLAGS += -Wno-unused-function
+EXTRA_CFLAGS += -Wno-error=restrict
+EXTRA_CFLAGS += -Wno-error=maybe-uninitialized
 #EXTRA_CFLAGS += -Wno-unused
 
 #EXTRA_CFLAGS += -Wno-uninitialized
diff --git a/net/netfilter/ipvs/Makefile b/net/netfilter/ipvs/Makefile
index 67f3f4389602..cfd6f4b15528 100644
--- a/net/netfilter/ipvs/Makefile
+++ b/net/netfilter/ipvs/Makefile
@@ -40,3 +40,5 @@ obj-$(CONFIG_IP_VS_FTP) += ip_vs_ftp.o
 
 # IPVS connection template retrievers
 obj-$(CONFIG_IP_VS_PE_SIP) += ip_vs_pe_sip.o
+
+CLFAGS_ip_vs_core.o += -Wno-error=maybe-uninitialized
-- 
2.26.2

